rossum/lib/api_client.py

```diff
@@ -219,29 +219,60 @@ def get_paginated(
         query: Optional[Dict[str, Any]] = None,
         *,
         key: str = "results",
+        sideloads: Optional[Iterable[APIObject]] = None,
     ) -> Tuple[List[Dict[str, Any]], int]:
+        if query is None:
+            query = {}
+
+        if sideloads is not None:
+            query = query.copy()
+            query["sideload"] = ",".join(str(sideload) for sideload in sideloads)
+
         response = self.get(path, query)
         response_dict = response.json()
 
-        res = response_dict[key]
+        res = response_dict
```

Comment:
This is a potentially dangerous change - can you please add tests for scenario where other key than `results` is being passed, to make sure it won't break anything? The typical scenario includes getting aggregated results from `documents` or `content` (rather than `results`) when listing all annotations.

Reply:
Hmm... I see now. I assumed the `get_paginated` is not part of the "official" API (how silly me) but if anyone can call it with anything that is unfortunate 😭. I was also led astray by [Official documentation](https://api.elis.rossum.ai/docs/) which suggests sideloading is supported only for `annotations` (oh, how silly me again)...

Covering listing of all objects with all kind of sideloads will take a while... May I leave this PR open, or should I close it until it is in more mature form?

Reply:
I haven't surrendered, yet! Last commit shows how sideloading could be done by slight extension of `APIObject` interface. The sideloading of annotations (with all supported sideloads) could look like this in the code:
```python
client.get_annotations(queue=1, status=["exported"], sideloads=[AUTOMATION_BLOCKERS, CONTENT("document_id", "sender_vat_id"), DOCUMENTS, MODIFIERS])
```
If it is considered 🆗 ... I would add missing tests for it and tidy it up.

Reply:
I am not sure about creating separate `ApiObject`s for this, as the purpose of the `APIObject` in the code is mainly to refer to a specific object as refered to within the `API Reference` section of our API docs. And modifiers, automation blockers and content are not API objects. I think that the parameter could simply be `sideloads: Optional[Iterable[str]] = None,` instead of `sideloads: Optional[Iterable[APIObject]] = None,` and then it would be up to the user to provide the correct one. Besides the fact mentioned above, this will make the code maintainable for longer time - as we cannot track every time a new sideload object is added to the Rossum API. Yet, the user will still have an option to use the `APIObject` to pass value to the parameter, avoiding an error, right?